azerothcore/azerothcore-wotlk
# feat(Scripts/Commands): Item restoration command

## Changes Proposed:
-  New command: `.item restore list <playerName>`
-  New command: `.item restore <restoreId> <playerName>`
-  Move `.itemmove` wip command into this new context

One will need to retrieve the id from the `recovery_Table` in order to restore the item. 
That's meant to be achieved by using the restore list command, then using the given id (not item id) in the restore command

This is for a more robust systematical performance and to easier adapt it to SOAP

## Tests Performed:

- Build works
- SQL queries gets applied, can get re-applied without errors
- Tested ingame
- Tested through console

## How to Test the Changes:


1. Enable either `ItemDelete.Method` or `ItemDelete.Vendor` in `worldserver.conf`

Make sure to pay attention to the `ItemDelete.Quality `and `ItemDelete.level` configs incase you want to try selling/deleting some lvl 5 milk or something

2. Restart worldserver if it's already on
3. Login, buy something from a vendor, sell it or destroy it
4. Run `.item restore list <characterName>`
5. Run `.item restore <id from list> <characterName>`
6. Visit mailbox to see a mail returning the item by the postmaster

You can also do the same steps in console, meaning you don't have to be online to run the commands or to get a mail sent to you.

## Known Issues and TODO List:


- Garbage recycle _recovery_items_ table, like a config setting expiration time before it can't be recovered any longer


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
